### PR TITLE
DataGridFilter via URLQuery does not work when DataGrid has key and property attributes configured

### PR DESCRIPTION
**Pull request by KremnevDmitry**
See: #3243

### CODE REVIEW

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/DataGridFilterUrlQueryParametersBinder.java [hunk 1]
```diff
@@ -21,7 +21,12 @@
 import com.vaadin.flow.component.Component;
 import com.vaadin.flow.component.grid.Grid;
 import com.vaadin.flow.router.QueryParameters;
+import com.vaadin.flow.shared.Registration;
```
Comment (glebfox) [inline jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/DataGridFilterUrlQueryParametersBinder.java:24]: unused imports

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/editor/DataGridEditorImpl.java [hunk 2]
```diff
@@ -412,7 +415,15 @@ public void initColumnDefaultEditorComponent(String property) {
     @Override
     public void setColumnEditorComponent(String property,
                                          Function<EditComponentGenerationContext<T>, Component> generator) {
-        Grid.Column<T> column = getGrid().getColumnByKey(property);
+        if (!(getGrid() instanceof EnhancedDataGrid)) {
+            throw new IllegalStateException(getGrid().getClass().getSimpleName() +
+                    " items is null or does not implement " + EntityDataUnit.class.getSimpleName());
```
Comment (glebfox) [inline jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/editor/DataGridEditorImpl.java:420]: items?

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/editor/DataGridEditorImpl.java [hunk 3]
```diff
@@ -412,7 +415,15 @@ public void initColumnDefaultEditorComponent(String property) {
     @Override
     public void setColumnEditorComponent(String property,
                                          Function<EditComponentGenerationContext<T>, Component> generator) {
-        Grid.Column<T> column = getGrid().getColumnByKey(property);
+        if (!(getGrid() instanceof EnhancedDataGrid)) {
```
Comment (glebfox) [inline jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/editor/DataGridEditorImpl.java:418]: Don't touch Editor!

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/DataGridFilterUrlQueryParametersBinder.java [hunk 4]
```diff
@@ -134,8 +142,12 @@ protected void applyPropertyFilterParameters(List<String> params) {
             String propertyString = parameterString.substring(0, separatorIndex);
             String property = urlParamSerializer.deserialize(String.class,
                     filterUrlQueryParametersSupport.restoreSeparatorValue(propertyString));
+            MetaClass entityMetaClass = ((ContainerDataGridItems<?>) grid.getDataProvider()).getEntityMetaClass();
 
-            DataGridColumn<?> column = (DataGridColumn<?>) grid.getColumnByKey(property);
+            DataGridColumn<?> column = grid.getColumnByMetaPropertyPath(metadataTools.resolveMetaPropertyPath(
```
Comment (glebfox) [inline jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/DataGridFilterUrlQueryParametersBinder.java:147]: Check case if we have several columns with different keys but bound to the same entity property

**Review by glebfox — CHANGES_REQUESTED** [review]

**Comment by KremnevDmitry** [comment]
Implementation will be changed

**Pull request closed without merging**